lib/internal/loader/ModuleRequest.js

```diff
@@ -36,10 +37,15 @@ loaders.set('esm', async (url) => {
 
 // Strategy for loading a node-style CommonJS module
 loaders.set('cjs', async (url) => {
+  const pathname = internalURLModule.getPathFromURL(new URL(url));
+  const module = CJSModule._cache[pathname];
+  if (module && module.loaded) {
+    const ctx = createDynamicModule(['default'], url, undefined);
+    ctx.reflect.exports.default.set(module.exports);
+    return ctx;
+  }
   return createDynamicModule(['default'], url, (reflect) => {
     debug(`Loading CJSModule ${url}`);
```

Comment:
How come `reflect` isn't used inside the `createDynamicModule` callback for `'cjs'`:
```js
reflect.exports.default.set(exports);
```
Like it is for others and the cached case above 👆 *(I'm not familiar with how things work with the loader)*

Reply:
using _load performs that right about here
https://github.com/nodejs/node/blob/317f596bab62e4c596e73c10c2d8c007b562a904/lib/module.js#L580

Reply:
@devsnek Ah, any way that could be tidied up so that all the `reflect.exports.default.set` is done near each other or at least in the same file?

Reply:
i think this is already as tidy as it will get, but i think a comment above line 49 would go a long way towards explaining what is going on. personally i had no idea how line 49 worked until when you asked because i decided it was time to figure it out.